torch/export/exported_program.py

```diff
@@ -224,6 +229,31 @@ def _override_composite_implicit_decomp(cia_ops_to_callable, safe=True):
                 decomp_callable
             )
 
+        # [NOTE] Directly registering fake tensor rule to CIA ops
```

Comment:
maybe just a nit that's helpful to me, but I think it would be useful to mention something like:

There are effectively 3 places where an op register tensor propagation logic:

(1) as a CompositeImplicitAutograd decomposition (above FakeTensorMode)
(2) as a FakeTensorMode registration
(3) as a `DispatchKey::Meta` registration

During export, we nub out most CIA ops to return `NotImplemented` to avoid decomposing them during tracing. To recover the existing shape propagation behavior, we register these CIA decomps directly as FakeTensorMode rules as well.

A valid question is: all CIA ops are also registered to the Meta key, so why can't we just rely on running the meta tensor kernels? The issue is that when FakeTensorMode runs a meta tensor kernel, the devices of all inputs have been nubbed out to return "meta". If we have any CIA op impls that need to branch on device, we need to register them as FakeTensorMode rules to preserve any device-specific behavior.